server/store/session.go

```diff
@@ -0,0 +1,170 @@
+package store
+
+import (
+	"strings"
+
+	pluginapi "github.com/mattermost/mattermost-plugin-api"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/server/config"
+	"github.com/mattermost/mattermost-plugin-apps/utils"
+)
+
+type SessionStore interface {
+	Get(appID apps.AppID, userID string) (*model.Session, error)
+	ListForApp(appID apps.AppID) ([]*model.Session, error)
+	ListForUser(userID string) ([]*model.Session, error)
+	Save(appID apps.AppID, userID string, session *model.Session) error
+	Delete(appID apps.AppID, userID string) error
+	DeleteAllForApp(appID apps.AppID) error
+}
+
+type sessionStore struct {
+	*Service
+}
+
+var _ SessionStore = (*sessionStore)(nil)
+
```

Comment:
Many things in this file seem less than optimal in term of performance. Is there any plan for these on short-mid term?

Reply:
The `Get`  and `Save` operations are as performance as they can be without a cache. A cache should remove one RPC call, which is not a big deal as there are plenty of RPC calls being made. Do we really need a cache soon?

The `Delete` and `List` operations happen on rare occasions like uninstalling an app. IMO, they can be slow.